File: apps/ios/nearby/App/AccountRefresh.swift
import Foundation

/// App-wide signal that the account's on-chain state changed (a send or consolidation executed), so
/// balance- and activity-backed screens should refresh. Posted by the gasless write actions and
/// observed by the Home and Activity view models.
enum AccountRefresh {
  static let didChange = Notification.Name("nearby.accountDidChange")

  /// Announce that the account changed.
  static func post() {
    NotificationCenter.default.post(name: didChange, object: nil)
  }

  /// An async stream of change events, for view models to `for await` in their lifecycle task.
  static var events: NotificationCenter.Notifications {
    NotificationCenter.default.notifications(named: didChange)
  }
}

File: apps/android/app/src/main/java/com/variance/nearby/core/AccountRefresh.kt
package com.variance.nearby.core

import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.SharedFlow

/**
 * App-wide signal that the account's on-chain state changed (a send or consolidation executed), so
 * balance- and activity-backed screens should refresh. Posted by the gasless write actions and
 * observed by the Home and Activity view models.
 */
object AccountRefresh {
    private val _events = MutableSharedFlow<Unit>(extraBufferCapacity = 16)
    val events: SharedFlow<Unit> = _events

    /** Announce that the account changed. */
    fun post() {
        _events.tryEmit(Unit)
    }
}
